code/modules/antagonists/cult/cult_items.dm

```diff
@@ -875,26 +875,26 @@
 	ADD_TRAIT(src, TRAIT_NODROP, CULT_TRAIT)
 
 
-/obj/item/blood_beam/afterattack(atom/A, mob/living/user, flag, params)
+/obj/item/blood_beam/afterattack(atom/A, mob/living/user, proximity_flag, clickparams)
 	. = ..()
 	if(firing || charging)
 		return
-	var/C = user.client
-	if(ishuman(user) && C)
-		var/list/angle_vector = calculate_projectile_angle_and_pixel_offsets(user, params)
-		angle = angle_vector[1]
+	if(ishuman(user))
+		angle = Get_Angle(user, A)
 	else
 		qdel(src)
 		return
 	charging = TRUE
 	INVOKE_ASYNC(src, .proc/charge, user)
-	if(do_after(user, 90, target = user))
+	if(do_after(user, 9 SECONDS, target = user))
 		firing = TRUE
-		INVOKE_ASYNC(src, .proc/pewpew, user, params)
+		ADD_TRAIT(user, TRAIT_IMMOBILIZED, CULT_TRAIT)
+		INVOKE_ASYNC(src, .proc/pewpew, user, clickparams)
 		var/obj/structure/emergency_shield/cult/weak/N = new(user.loc)
-		if(do_after(user, 90, target = user))
+		if(do_after(user, 9 SECONDS, target = user))
 			user.Paralyze(40)
 			to_chat(user, "<span class='cult italic'>You have exhausted the power of this spell!</span>")
```

Comment:
honestly, it would probably be an improvement if we just removed the 4 second hardstun for letting the beam go for its full duration. we don't need to punish people for having fun with their absurdly impractical doom laser for too long, and blood beam probably deserves the (very minor) buff.

dropping it would also mean that we wouldn't need to do this immobilization trait stuff.

Reply:
It can still cause bugs by starting a bloodbeam then just leaving before it actually ends, even if the stun wasn't there.

Reply:
what?

Reply:
Starting a bloodbeam, being in the middle of shooting the beams, then just walking out as you're still beaming.

Reply:
I'm saying that we should just make allowing people to end the blood beam at any time be the intended functionality and remove the hardstun that punishes them for firing the beam for its full duration.